packages/component-library/src/Button/index.tsx

```diff
@@ -0,0 +1,129 @@
+import clsx from "clsx";
+import type { ComponentProps, ComponentType, SVGProps } from "react";
+import { Button as BaseButton, Link } from "react-aria-components";
+
+type Icon = ComponentType<SVGProps<SVGSVGElement>>;
+
+type OwnProps = {
+  variant?: "primary" | "secondary" | "solid" | "outline" | "ghost";
+  size?: "xs" | "sm" | "md" | "lg";
+  rounded?: boolean;
+  children?: string;
+  beforeIcon?: Icon;
+  afterIcon?: Icon;
+};
+
+export type ButtonProps = Omit<
+  ComponentProps<typeof BaseButton>,
+  keyof OwnProps
+> &
+  OwnProps;
+
+export const Button = (props: ButtonProps) => (
+  <BaseButton
+    {...props}
+    {...dataAttributes(props)}
+    className={clsx(
+      baseClasses,
+
+      // Pending
+      "data-[pending]:data-[variant]:cursor-wait data-[pending]:data-[variant]:border-transparent data-[pending]:data-[variant]:bg-stone-200 data-[pending]:data-[variant]:text-stone-400 data-[pending]:data-[variant]:data-[focus-visible]:outline-stone-300 dark:data-[pending]:data-[variant]:bg-slate-600 dark:data-[pending]:data-[variant]:text-slate-400 dark:data-[pending]:data-[variant]:data-[focus-visible]:outline-slate-500",
+
+      props.className,
+    )}
+    {...props}
+  >
+    <ButtonInner {...props} />
+  </BaseButton>
+);
+
+export type ButtonLinkProps = Omit<
+  ComponentProps<typeof Link>,
+  keyof OwnProps
+> &
+  OwnProps;
+
+export const ButtonLink = (props: ButtonLinkProps) => (
+  <Link
+    {...props}
+    {...dataAttributes(props)}
+    className={clsx(baseClasses, props.className)}
+  >
+    <ButtonInner {...props} />
+  </Link>
+);
+
+const Icon = ({ icon: IconComponent }: { icon: Icon }) => (
+  <IconComponent className="inline-block size-6 group-data-[size=sm]:size-5 group-data-[size=xs]:size-4" />
+);
+
+const ButtonInner = (props: OwnProps) => (
+  <>
+    {"beforeIcon" in props && <Icon icon={props.beforeIcon} />}
+    {"children" in props && props.children !== "" && (
```

Comment:
Out of curiosity, isn't it possible for a component to return null in some cases?
```
if (...) {
   return null;
}

if (...) {
   return null;
}

return <Text> Admin </Text>;
```

I think you are using the `&&` pattern in the code, but I think it can happen. Not sure if there is any specific eslint rule for that or maybe I'm missing sth here.

Reply:
In this case no because the type of `children` is `children?: string`: https://github.com/pyth-network/pyth-crosschain/pull/2042/files#diff-4f427b4c6e5288a1739a1654ff79451ca88da6f7cb92a0fc95a866d82c27e9b9R11